app/controllers/sub_space_booking_controller.rb

```diff
@@ -219,11 +323,48 @@ def decline
       )
     booking.booking_status_id = BookingStatus::DECLINED.id
     booking.save
-    redirect_to admin_sub_space_booking_index_path,
+    redirect_to sub_space_booking_index_path(anchor: "booking-admin-tab"),
                 notice:
                   "Booking for #{SubSpaceBooking.find(params[:sub_space_booking_id]).sub_space.name} declined successfully."
   end
 
+  def publish
+    unless current_user.admin?
```

Comment:
Create a method that checks if the user is an admin and add a before_action on the controller instead of doing it in multiple spots, it will be easier to read and save lines